### ISSUE TITLE
Hunter feign death not reliably allowing use of out of combat activities such as traps/food/item swapping in boss encounters

### ISSUE THREAD

**Issue by whabt** [LightsHope/server#886]
The behavior described below began around the time of patch 1.8 was implemented on Darrowshire. All information in this report was collected on Darrowshire, Alliance faction.  I was unable to find a duplicate/similar bug report.  Prior to this issue, feign death dropped combat and ooc abilities/activities (traps, food, gear swapping) lit up, became usable, and remained so until the hunter canceled feign and engaged/was aggrod by the encounter normally, or the hunter's pet attacked or was attacked by an NPC while feigned.

During boss encounters, most notably during Razorgore and Nefarion when hunter traps are widely used during the encounter for crowd control (but also noticed during some other encounters in BWL and MC), the hunter skill Feign Death does not seem to be reliably allowing the player to trigger out of combat activities. Upon skill activation, during a boss encounter, the animation plays and icons for out of combat activities (food, traps, etc) will light up momentarily but then immediately go dark, preventing their use.  This happens without any other player activity, and whether pets are dismissed or summoned.  Threat is dropped normally (and does not return), and while feign is active NPCs will not attack the hunter, even after all other player characters are dead.  Feign appears to be working except for the inability to use ooc abilities while feigned.  I have only seen it occur during boss encounters.  Frequency is nearly all the time; very rarely a trap or gear swap will go through as normal, but the usual result is the "You are in combat" error message.  I was unable to successfully drop a trap at all during any bosses on our most recent bwl clear (razor p1, nef p1/p3).  If there is an actual window to use any ooc abilities, it is incredibly short; neither macro use nor addons (queueing a gear swap up in itemrack) allow you to take advantage of it reliably.

Add-ons do not seem to make a difference in this behavior.  I have disabled all addons, as well as cleared my wdb and wtf folders before our most recent razorgore kill and the behavior persisted.  All intentional testing I have done was during the Razorgore encounter for consistency, and wdb and wtf folders were cleared prior to my last Razorgore kill.  Behavior did not improve.

This behavior is different from feign death resist - resists appear to be working normally. Imp Feign Death does not appear to affect the frequency of the ooc ability bug (though it is hard to really tell since the ooc bug is so common). Talent specialization does not appear to make a difference.  

Thank you.

**Comment by Zarant** [comment]
Raid bosses usually do a combat pulse on the entire zone every ~3s and this combat pulse ignores FD or vanish and put you back in combat right away. If you feign a split second before the combat pulse occurs, chances are that you can't lay any traps or switch gear.

**Comment by whabt** [comment]
The behavior is too consistent to be this.  I would have to be incredibly lucky in order to feign a few tenths of a second ahead of the combat pulse every single time.  It's too reliable to randomly be running into the combat pulse, assuming it's only every few seconds.  Out of combat abilities light up and then fade instantly after feigning death almost 100% of the time during the Razorgore encounter (and others as well).

**Comment by Zarant** [comment]
I just did a quick glance over the nefarian script and I can see that every time nef switch targets he puts the whole zone in combat. There are few other things that can trigger this combat pulse, I suspect that on those 2 fights you have multiple adds triggering it, so the chance of you getting a trap in is quite low.

**Comment by whabt** [comment]
It is now impossible to reliably drop a hunter trap in combat during boss encounters. Not just nef, but most boss encounters. Trinket swaps aren't working during fights like ebonroc when only one NPC is in combat. 

Feign death did not previously put you into combat before canceling it without performing some action (typically a pet attack) that would put you into combat. Not even during bosses. Currently combat is regained without player interaction and so fast (effectively instantly) that no out of combat action can reliably taken after feigning. This is not in line with how traps worked in retail nor is it in line with how traps worked prior to whatever change happened to cause this behavior. The chance for in combat trapping to fail has always been tied to both the traps chance to resist and feign deaths chance to resist, it already has twice the opportunity to resist than other spells.

The behavior of falling into combat while feigned and not actively attacking or attacking with your minion is broken.

**Comment by GuybrushGit** [comment]
This is intentional, its a result of the combat pulse we recently implemented. You can still feign and trap, you just have to be fast.

**Comment by Zarant** [comment]
Any damage you do after using feign death will also put you in combat, if you feign a split second after an arrow goes off you will most likely be put back in combat.

**Comment by whabt** [comment]
I'm aware of the arrows still in flight putting you back into combat.

As for having to be fast, how fast exactly? Because queued itemrack gear swaps aren't fast enough, FD/trap macro mousewheel spam isn't fast enough.  It's not just difficult, it's nigh impossible to perform a basic class ability during boss encounters, and it's probably safe to say that hunter trap use is a pretty established strategy during these fights.

I'm not suggesting that the combat pulse should go or anything. But as it is now, it's way too fast to use ooc hunter abilities after a feign. Is there a reason it's so fast? I mean, rezzes and summons can be stopped with a combat pulse every 5 seconds or so can't they? Is it to prevent drinking after taking a combat rez or something? 

At any rate, if it's the combat pulse, it's way too aggressive to use traps, swap trinkets/gear, drink, etc, during fights. These were all functional a little while ago and were never prevented during boss fights in retail.

I'll try to attach a video after we raid next showing how fast abilities grey out during those fights, but it is reliably too fast to trigger any ooc ability manually/with a macro. Load up razorgore and try to drop some traps in p1. It just doesn't work reliably.  If you could take a look at it in practice I'd be appreciative.

**Comment by Zarant** [comment]
Here are some video footage from retail:

https://www.youtube.com/watch?v=OO8D4elx7Us
At the 40s mark you can see that there is a 2 to 3 second grace period to lay down a trap after a feign, which is consistent with the ~3s combat pulse implemented on this server. But that was the only case where the hunter was OOC for more than 2s, which may indicate that it's working as intended, considering that the grace period is seemingly random. 

Additional footage:
https://www.youtube.com/watch?v=kYrgM_WmaO8
https://www.youtube.com/watch?v=ofnd3J6sddk

I played a hunter on retail and i don't remember being so unreliable like that, and from the footage I just linked, with about ~10 casts across those 3 videos, the combat pulse never affected the hunter while feigned, although it may be just a coincidence, there is not enough data to conclude anything.

It boils down to a design choice at this point, since i doubt you will find any reliable data or decent sample size to judge if this is blizzlike or not.
IMO, the right thing to do is consider the player dead by the whole duration of the feign death buff, or in other words, make sure the combat pulse doesn't affect players with the feign death buff.

**Comment by whabt** [comment]
I agree that that is probably the right way to handle it. I also played hunter in retail and remember feign always leaving the hunter fully out of combat, with pulses affecting you only after you stood up. 

Also, just to be extra clear: combat is not being regained a second or two after getting up, it's not even instantly upon getting up. It's instantly after casting feign death. You're back in combat before the animation even finishes. The trap cast fails because you are in combat.

**Comment by whabt** [comment]
Didn't mean to close it, accidentally hit the wrong button on my phone.

**Comment by knasen88** [comment]
if you dont call back your pet and it autohits it will put you in combat again

**Comment by whabt** [comment]
@knasen88 I'm aware, this isn't that.

**Comment by staplerx300** [comment]
I have also noticed this. Whatever Guybrush is doing clearly isn't working for everyone. I only get half my mid-combat traps through on Nefarian now.

**Comment by ifyousayso** [comment]
From what I can tell at https://www.youtube.com/watch?v=OO8D4elx7Us , the hunter is not in combat right before 0:40 and gets in combat right before 0:44. That's 4 seconds. But even if it's not always 4, how about this idea:
I have an addon that shows energy updates for rogues. I'm guessing it's some kind of update sent by the server? What if you use that same update to put PCs in combat and let abilities that break combat (such as Feign Death) make a character skip the very next combat trigger? That means that after a successful Feign Death, a hunter would have a minimum of 2 but less than 4 seconds to set a trap. (I'm pretty sure it runs every two seconds.)

**Comment by Zarant** [comment]
Found more video footage:
https://www.youtube.com/watch?v=URIOZLiLJ-g
You can see the hunter using Feign death at the following:
2:11 
2:35
3:41
4:05
In all those cases the time required to get back in combat was over a GCD.

At 4:43 combat is regained half a second after cancelling feign death, but still had enough time set a trap.
At 6:01 the hunter exit feign death after casting a trap, takes exactly 3s to get back into combat.

This other video is also interesting, from a rogue PoV: https://www.youtube.com/watch?v=yn5IHvHFDtc
At 6:01 he casts vanish, drop out of combat and then hit Nefarian again at 6:04, but notice that the video in question is sped up by roughly 1.5 times. He remained out of combat for the whole duration of the vanish buff, roughly 4-5 seconds, which is longer than the ~3s combat pulse.
